icechunk-python/tests/test_xarray.py

```diff
@@ -50,20 +50,23 @@ def create_test_data(
 
 
 @contextlib.contextmanager
-def roundtrip(data: xr.Dataset) -> Generator[xr.Dataset, None, None]:
+def roundtrip(
+    data: xr.Dataset, allow_pickling: bool = False
+) -> Generator[xr.Dataset, None, None]:
     with tempfile.TemporaryDirectory() as tmpdir:
         repo = Repository.create(local_filesystem_storage(tmpdir))
         session = repo.writable_session("main")
-        to_icechunk(data, store=session.store, mode="w")
 
-        # if allow_distributed_write:
-        #     with session.allow_distributed_write():
-        #       to_icechunk(data, store=session.store, mode="w")
-        # else:
-        #     to_icechunk(data, store=session.store, mode="w")
+        if allow_pickling:
+            with session.allow_pickling():
+                to_icechunk(data, store=session.store, mode="w")
+                with xr.open_zarr(session.store, consolidated=False) as ds:
```

Comment:
this is not so nice now. 

The distributed-read piece needs to be within the context manager now, because we can't (easily) set read-only status when unpickling. We could do so I think, by creating a brand new Session with branch=None. Let's discuss tomorrow..